.github/workflows/publish.yaml

```diff
@@ -0,0 +1,46 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - main
+      - track/**
+  schedule:
+    - cron: '0 8 * * THU'
+
+jobs:
+  charm:
+    name: Publish Charm
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        set -eux
+        sudo snap install charmcraft --classic --channel=latest/candidate
+        sudo snap install yq
+
+    - name: Charm
+      env:
+        CHARMCRAFT_CREDENTIALS: ${{ secrets.CHARMCRAFT_CREDENTIALS }}
+        CHARMCRAFT_DEVELOPER: "y"
```

Comment:
We can actually remove this line. I forgot to remove it from `dex-auth`, but it's only necessary if you've installed `charmcraft` via pip.

Reply:
+1, makes sense.